### ISSUE TITLE
write midi error

### ISSUE THREAD

**Issue by YoshikawaMasashi** [craffel/pretty-midi#114]
this code makes error.(Python3)

```
import pretty_midi

pm = pretty_midi.PrettyMIDI(resolution=960, initial_tempo=120) 
instrument = pretty_midi.Instrument(0) 

note_number = pretty_midi.note_name_to_number('G4')
note = pretty_midi.Note(velocity=100, pitch=note_number, start=0, end=1) 

instrument.notes.append(note)
pm.instruments.append(instrument)
pm.write('test.mid') 
```

but if change this code
```
note = pretty_midi.Note(velocity=100, pitch=note_number, start=0, end=1) 
```
to
```
note = pretty_midi.Note(velocity=100, pitch=note_number, start=1e-20, end=1) 
```
I have no error.

the cause of this error
time_to_tick(0) return 0(np.int64) not 0(int).

the cause of np.int64 
np.searchsorted in time_to_tick return np.int64

sorry bad english.

**Comment by craffel** [comment]
Interesting, thanks for the heads up.  This is an issue with how `mido` does type-checking.  If `a = np.int64(10)`, then in Python 2, `isinstance(a, int) = True` and in Python 3, `isinstance(a, int) = False`.  A better check, I think, would be duck-type-style, like
```Python
def check_time(time):
    valid = True
    try:
        int(time)
    except:
        valid = False
    try:
        float(time)
    except:
        valid = False
    if not valid:
        raise TypeError('time must be an integer or float')
```

Do you want to create an issue on `mido` requesting this change, or should I?

**Comment by calclavia** [comment]
I'm running into the same problem. Making a PR to MIDO

**Comment by ejhumphrey** [comment]
just stepped in this walking some folks through a MIDI tutorial with the version on PyPI, is there anything that has to happen for this to propagate through to be pip-installable?

**Comment by craffel** [comment]
@olemb said he pushed this change into mido version 1.1.21, is that the version available on PyPI/the one your folks had installed?

**Comment by ejhumphrey** [comment]
`pip install pretty_midi` caused issues, `pip install git+git://github.com/craffel/pretty-midi` did not ... so I don't think it's a mido issue? I think?

**Comment by calclavia** [comment]
It should work after you update MIDO `pip install mido --upgrade`

**Comment by craffel** [comment]
> pip install pretty_midi caused issues

What issues?  If PIP didn't install `pretty_midi`, that's a separate issue.

**Comment by ejhumphrey** [comment]
apologies for the confusion -- pip install pretty_midi was struggling from
the same issue that started this thread, pip installing from github source
doesn't.

I can look into the midi upgrade to see if that really the issue


On Sat, Jan 28, 2017 at 4:39 PM Colin Raffel <notifications@github.com>
wrote:

pip install pretty_midi caused issues

What issues? If PIP didn't install pretty_midi, that's a separate issue.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/craffel/pretty-midi/issues/114#issuecomment-275876465>,
or mute the thread

**Comment by craffel** [comment]
> apologies for the confusion -- pip install pretty_midi was struggling from
the same issue that started this thread, pip installing from github source
doesn't.

Hmm, the issue in this thread was caused by https://github.com/olemb/mido/issues/78 , so should be resolved assuming you are running mido 1.1.21.  There have been a few others commits to `pretty_midi` since the version on PyPI, if not having them is causing an issue for you I can push a new release with the latest commits.
